pkg/version/version.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package version
+
+import (
+	"fmt"
+	"runtime"
+)
+
+type Info struct {
```

Comment:
why don't we use the https://github.com/kubernetes/apimachinery/blob/master/pkg/version/types.go#L22, which can be vendored instead of defining it in our package.

Reply:
This repo already vendors apimachinery:

https://github.com/kubernetes-sigs/federation-v2/tree/master/vendor/k8s.io/apimachinery

Reply:
The reason I did put in federation specific structure is that, I did not find some of the fields meaningful for us, viz `Major` and `Minor`. I find the details repetitive, the same info is part of `Version` and as mentioned in the comment where the global var is declared [here](https://github.com/irfanurrehman/federation-v2/blob/0a83ed76eeb93b5469620205681e0a6523144dd8/pkg/version/base.go#L29), the version info should be a tag in the repo (ideally on master). This will ensure we do not have multiple places tracking versions.